Framework/DataHandling/src/LoadPSIMuonBin.cpp

```diff
@@ -112,22 +157,53 @@ void LoadPSIMuonBin::exec() {
     outputWorkspace->mutableX(specNum) = m_xAxis;
     outputWorkspace->mutableY(specNum) = m_histograms[specNum];
     outputWorkspace->mutableE(specNum) = m_eAxis[specNum];
+    outputWorkspace->getSpectrum(specNum).setDetectorID(specNum + 1);
   }
 
   assignOutputWorkspaceParticulars(outputWorkspace);
 
+  // Set up for the Muon PreProcessor
   setProperty("OutputWorkspace", outputWorkspace);
+
+  auto largestBinValue =
+      outputWorkspace->x(0)[outputWorkspace->x(0).size() - 1];
+
+  auto firstGoodDataSpecIndex = static_cast<int>(
+      *std::max_element(m_header.firstGood, m_header.firstGood + 16));
+  setProperty("FirstGoodData", outputWorkspace->x(0)[firstGoodDataSpecIndex]);
+
+  // Since the arrray is all 0s before adding them this can't get min
+  // element so just get first element
+  auto lastGoodDataSpecIndex = static_cast<int>(m_header.lastGood[0]);
+  setProperty("LastGoodData", outputWorkspace->x(0)[lastGoodDataSpecIndex]);
+
+  double timeZero = 0.0;
+  if (m_header.realT0[0] != 0) {
+    timeZero = m_header.realT0[0];
+  } else {
+    timeZero = static_cast<double>(m_header.integerT0[0]);
+  }
+
+  // If timeZero is bigger than the largest bin assume it refers to a bin's
+  // value
+  if (timeZero > largestBinValue) {
+    setProperty("TimeZero",
+                outputWorkspace->x(0)[static_cast<int>(std::floor(timeZero))]);
+  } else {
+    setProperty("TimeZero", timeZero);
+  }
 }
 
 std::string LoadPSIMuonBin::getFormattedDateTime(std::string date,
                                                  std::string time) {
-  std::map<std::string, std::string> months{
-      {"JAN", "01"}, {"FEB", "02"}, {"MAR", "03"}, {"APR", "04"},
-      {"MAY", "05"}, {"JUN", "06"}, {"JUL", "07"}, {"AUG", "08"},
-      {"SEP", "09"}, {"OCT", "10"}, {"NOV", "11"}, {"DEC", "12"}};
-  return "20" + date.substr(7, 2) + "-" +
-         months.find(date.substr(3, 3))->second + "-" + date.substr(0, 2) +
-         "T" + time;
+  std::string year;
+  if (date.size() == 11) {
+    year = date.substr(7, 4);
```

Comment:
why these substrings?

Reply:
That's where the year is located in the date string, I didn't add a comment because I thought the variable name was clear enough, should I add one?

Reply:
just an example date would suffice.